app/src/org/commcare/activities/FormEntryActivity.java

```diff
@@ -909,7 +900,19 @@ protected void onPause() {
         }
         TextToSpeechConverter.INSTANCE.stop();
 
-        unregisterReceiver(pendingSyncAlertBroadcastReceiver);
+        attemptToUnregisterBroadcastReceiver(pendingSyncAlertBroadcastReceiver);
+    }
+
+    private void attemptToUnregisterBroadcastReceiver(BroadcastReceiver broadcastReceiver){
+        if (broadcastReceiver != null) {
+            try {
+                unregisterReceiver(broadcastReceiver);
+            } catch (IllegalArgumentException e) {
+                // Thrown when given receiver isn't registered.
+                Logger.log(LogTypes.TYPE_ERROR_ASSERTION,
```

Comment:
we should log this using `logger.exception()`

Reply:
I'd normally agree with this, but reviewing the [baseline docs](https://developer.android.com/reference/android/content/Context#unregisterReceiver(android.content.BroadcastReceiver)), it's clear that there are only two scenarios where this gets thrown and both involve a semantically clear outcome (the receiver was not registered before, or was already cleared). 

As such, I don't think necessary to log as an exception directly, which I think of as more of a clearing house for things we believe are (or are might be) actual errors. To me this is more like a NumberFormatException catch, which reflects a known condition using the catch callpath.